File: client-python3/AnimationSender.py
import socket


class AnimationSender(object):

    def __init__(self, ip_address, port_num):
        self.address = ip_address
        self.port = port_num
        self.connection = socket.socket()

    def start(self):
        self.connection = socket.create_connection((self.address, self.port))

    def end(self):
        self.connection.close()

    def send_animation(self, animation_json):
        json = bytearray(animation_json, "utf-8")
        self.connection.send(json)

File: client-ruby/lib/animation_sender.rb
require 'socket'
require_relative './animation_sender.rb'

class AnimationSender
  attr_accessor :address, :port

  def initialize(address, port)
    @address = address
    @port = port
  end

  def start
    @socket = TCPSocket.new @address, @port
  end

  def end
    @socket.close
  end

  def send_animation(animation)
    raise TypeError unless animation.is_a? AnimationData

    @socket.write animation.json
  end

end
